packages/api-client/src/views/Request/hooks/useOpenApiWatcher.ts

```diff
@@ -0,0 +1,208 @@
+import { useWorkspace } from '@/store'
+import { specDictionary } from '@/store/import-spec'
+import {
+  combineRenameDiffs,
+  diffToCollectionPayload,
+  diffToRequestPayload,
+  diffToSecuritySchemePayload,
+  diffToServerPayload,
+  diffToTagPayload,
+} from '@/views/Request/libs/live-sync'
+import { createExampleFromRequest } from '@scalar/oas-utils/entities/spec'
+import { createHash, fetchSpecFromUrl } from '@scalar/oas-utils/helpers'
+import { parseSchema } from '@scalar/oas-utils/transforms'
+import { useToasts } from '@scalar/use-toasts'
+import { useTimeoutPoll } from '@vueuse/core'
+import microdiff, { type Difference } from 'microdiff'
+import { watch } from 'vue'
+
+/** Live Sync polling timeout */
+const FIVE_SECONDS = 5 * 1000
+
+/**
+ * Hook which handles polling the documentUrl for changes then attempts to merge what is new
+ *
+ * TODO:
+ * - check lastModified or similar headers
+ * - speed up the polling when there's a change then slowly slow it down
+ */
+export const useOpenApiWatcher = () => {
+  const { toast } = useToasts()
+  const {
+    activeCollection,
+    activeWorkspace,
+    collectionMutators,
+    requests,
+    requestMutators,
+    requestExamples,
+    requestExampleMutators,
+    securitySchemes,
+    securitySchemeMutators,
+    servers,
+    serverMutators,
+    tags,
+    tagMutators,
+  } = useWorkspace()
+
+  /** Little toast helper */
+  const toastError = (type: string) =>
+    toast(
+      `[useOpenApiWatcher] changes to the ${type} were not applied`,
+      'error',
+    )
+
+  // Transforms and applies the diff to our mutators
+  const applyDiff = (d: Difference) => {
```

Comment:
This function is very large and very heavy. It needs to have at least some level of test coverage. 

I would suggest we pull the apply diff function out and provide the `StoreContext` as second parameter to the function. We use this pattern elsewhere in the store. This would allow separate unit testing of the apply diff function. 

The function can return a success boolean and error messages that can be toasted. 

I would also prefer an early return pattern over the massive list of else-ifs but thats more of a nit.

Reply:
yea so it started as one massive function, this is actually the third time it was broken out, was just avoiding having to mock mutators but I can do that